repair/row_level.cc

```diff
@@ -1325,6 +1327,43 @@ class repair_meta {
         // Clear gently to avoid stalls
         utils::clear_gently(row_diff).get();
     }
+public:
+    const std::unique_ptr<const locator::token_metadata>& get_tm_for_small_table_optimization_check(const locator::token_metadata* tm) {
+        if (_small_table_optimization_tm_calculated) {
+            return _small_table_optimization_tm;
+        }
+        if (!tm) {
+            return _small_table_optimization_tm;
+        }
+        std::unique_ptr<locator::token_metadata> tmptr;
+        if (_reason == streaming::stream_reason::bootstrap) {
+            auto myid = tm->get_my_id();
+            std::unordered_set<dht::token> bootstrap_tokens;
+            for (auto [token, host_id] : tm->get_bootstrap_tokens()) {
+                if (myid == host_id) {
+                    bootstrap_tokens.insert(token);
+                }
+            }
+            if (!bootstrap_tokens.empty()) {
+                // The node is a bootstrapping node
+                tmptr = std::make_unique<locator::token_metadata>(tm->clone_only_token_map().get());
+                tmptr->update_normal_tokens(bootstrap_tokens, myid).get();
```

Comment:
What is the goal of this manipulation, to create a tm as if the bootstrap was already finished?

Why wasn't it needed before?

Reply:
Yes, to get a tm as if the bootstrap is done. It is needed to filter out data that does not belong to the node which is  needed by the small table optimization. It is not needed before because we do not turn on small table optimization for repair used in bootstrap.